compiler/luci/pass/src/FoldShapePass.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FoldShapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+template <loco::DataType OutType> luci::CircleConst *shape_of(luci::CircleNode *input_tensor)
+{
+  auto name = input_tensor->name();
+  auto shape_status = input_tensor->shape_status();
+  auto rank = input_tensor->rank();
+  assert(name.length() > 0);
+  assert(shape_status == luci::ShapeStatus::VALID);
+  assert(rank > 0);
+
+  auto shape = input_tensor->graph()->nodes()->create<luci::CircleConst>();
+  shape->name(name + "_ConstShape");
+  shape->dtype(OutType);
+  shape->rank(1);
+  shape->dim(0).set(rank);
+
+  shape->size<OutType>(rank);
+  for (uint32_t i = 0; i < rank; i++)
+    shape->at<OutType>(i) = input_tensor->dim(i).value();
+
```

Comment:
You need to add an origin about this new operator that is gonna help developers know where the node came from.

Reply:
I understand what you mean, but I failed to find a method to set the origin 😥

Can I ask how to set the origin for the node?

Reply:
Please refer to below codes.

https://github.com/Samsung/ONE/blob/d0cff18b41b4eeb0d1a59bfb7790ab8007f3e033/compiler/luci/pass/src/ResolveCustomOpSplitVPass.cpp#L37C13-L37C13
```cxx
luci::add_origin(cloned, luci::get_origin(node));
```